services/brig/src/Brig/Index/Options.hs

```diff
@@ -188,29 +188,35 @@ restrictedElasticSettingsParser = do
       { _esConnection =
           localElasticSettings._esConnection
             { esServer = server,
-              esIndex = ES.IndexName (prefix <> "_test"),
+              esIndex = mkIndexName (prefix <> "_test"),
               esCredentials = mCreds,
               esCaCert = mCaCert,
               esInsecureSkipVerifyTls = verifyCa
             }
       }
 
-indexNameParser :: Parser ES.IndexName
+indexNameParser :: Parser String
 indexNameParser =
-  ES.IndexName . view packed
-    <$> strOption
-      ( long "elasticsearch-index"
-          <> metavar "STRING"
-          <> help "Elasticsearch Index Name."
-          <> value (view (_IndexName . unpacked) localElasticSettings._esConnection.esIndex)
-          <> showDefault
-      )
+  strOption
+    ( long "elasticsearch-index"
+        <> metavar "STRING"
+        <> help "Elasticsearch Index Name."
+        <> value
+          ( Text.unpack
+              ( ES.unIndexName (localElasticSettings._esConnection.esIndex)
+              )
+          )
+        <> showDefault
+    )
+
+mkIndexName :: String -> ES.IndexName
+mkIndexName = either (error "invalid index name") id . ES.mkIndexName . Text.pack
 
 connectionSettingsParser :: Parser ESConnectionSettings
 connectionSettingsParser =
   ESConnectionSettings
     <$> elasticServerParser
-    <*> indexNameParser
+    <*> fmap mkIndexName indexNameParser
```

Comment:
Why was this change needed? Why the `error` function in case of failed parsing?

Reply:
Yes, failing after parsing was indeed a bit weird.

I'm addressing this with https://github.com/wireapp/wire-server/pull/4261/commits/fcd0f1a8c79233ea5b2ae75e4ed1b8d32071063e

Could you please take a look?

Reply:
Looks good!